File: leetcode/950.js
// 950. Reveal Cards In Increasing Order

/**
 * @param {number[]} deck
 * @return {number[]}
 */
const deckRevealedIncreasing = (deck) => {
  deck.sort((a, b) => a - b);
  const ans = [];
  while (deck.length) {
    ans.unshift(deck.pop());
    ans.unshift(ans.pop());
  }
  ans.push(ans.shift());
  return ans;
};

File: leetcode/950.rb
# 950. Reveal Cards In Increasing Order


# @param {Integer[]} deck
# @return {Integer[]}

def deck_revealed_increasing(deck)
  deck.sort!
  result = []

  result.unshift(deck.pop)

  deck.reverse_each do |val|
    el = result.pop
    result.unshift(el)
    result.unshift(val)
  end

  result
end